pkg/dependencies/factory.go

```diff
@@ -144,6 +145,10 @@ type executablesConfig struct {
 	mountDirs          []string
 }
 
+type timeoutConfig struct {
```

Comment:
since there is only one timeout setting, maybe just call it `config`?
Up to you, no strong opinion